src/Npgsql/PostgresException.cs

```diff
@@ -271,6 +271,38 @@ public override bool IsTransient
         /// </summary>
         public NpgsqlStatement Statement { get; internal set; }
 
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            var builder = new StringBuilder(base.ToString());
+            builder.AppendLine().Append("Exception data:");
+
+            void AppendPropertyLine(string propertyName, object propertyValue)
+            {
+                if (propertyValue == null) return;
```

Comment:
The int properties will never be null, and IIRC 0 means "not provided" (but this may be worth confirming in the PostgreSQL docs and in the Npgsql sources). If that's the case, you probably want to have different overloads for strings and ints, and check for 0 to mean non-present